ICE: rustdoc:  panicked at 'Error''

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.

http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/

-->

### Code
`./src/test/ui/type-alias-impl-trait/not_well_formed.rs`
```Rust
#![feature(type_alias_impl_trait)]

fn main() {}

trait TraitWithAssoc {
    type Assoc;
}

type Foo<V> = impl Trait<V::Assoc>; //~ associated type `Assoc` not found for `V`

trait Trait<U> {}

impl<W> Trait<W> for () {}

fn foo_desugared<T: TraitWithAssoc>(_: T) -> Foo<T> {
    ()
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->


```
rustdoc 1.62.0-nightly (4ca19e09d 2022-04-19)
binary: rustdoc
commit-hash: 4ca19e09d302a4cbde14f9cb1bc109179dc824cd
commit-date: 2022-04-19
host: x86_64-unknown-linux-gnu
release: 1.62.0-nightly
LLVM version: 14.0.1
```

### Error output

```
<output>
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
error[E0220]: associated type `Assoc` not found for `V`
 --> ./src/test/ui/type-alias-impl-trait/not_well_formed.rs:9:29
  |
9 | type Foo<V> = impl Trait<V::Assoc>; //~ associated type `Assoc` not found for `V`
  |                             ^^^^^ associated type `Assoc` not found

thread 'rustc' panicked at 'Error', src/librustdoc/clean/mod.rs:1671:29
stack backtrace:
   0:     0x7f242c69d85d - std::backtrace_rs::backtrace::libunwind::trace::hd79e9b51bb0b02a3
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f242c69d85d - std::backtrace_rs::backtrace::trace_unsynchronized::hc4b2624d11f57391
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f242c69d85d - std::sys_common::backtrace::_print_fmt::h5b920b6df28041d5
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7f242c69d85d - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h01f2db48eea34166
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7f242c6f8efc - core::fmt::write::h743b8fce003c331c
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/core/src/fmt/mod.rs:1194:17
   5:     0x7f242c68f041 - std::io::Write::write_fmt::h55edc38b905db9b5
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/io/mod.rs:1655:15
   6:     0x7f242c6a0575 - std::sys_common::backtrace::_print::h72c54a6b7a86b7bf
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7f242c6a0575 - std::sys_common::backtrace::print::h8b541992f5fa33c9
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7f242c6a0575 - std::panicking::default_hook::{{closure}}::h47e8a61e5844dea4
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/panicking.rs:295:22
   9:     0x7f242c6a01e9 - std::panicking::default_hook::h65ae1796882c178c
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/panicking.rs:314:9
  10:     0x7f242cecc5a1 - rustc_driver[caca827775d68846]::DEFAULT_HOOK::{closure#0}::{closure#0}
  11:     0x7f242c6a0d46 - std::panicking::rust_panic_with_hook::h1c3eee211b989bad
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/panicking.rs:702:17
  12:     0x7f242c6a0b09 - std::panicking::begin_panic_handler::{{closure}}::h653627205f5b2cdc
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/panicking.rs:586:13
  13:     0x7f242c69dd14 - std::sys_common::backtrace::__rust_end_short_backtrace::h36d845a914b6aae7
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/sys_common/backtrace.rs:138:18
  14:     0x7f242c6a0879 - rust_begin_unwind
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/panicking.rs:584:5
  15:     0x7f242c665bd3 - core::panicking::panic_fmt::hb6389d787a80a806
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/core/src/panicking.rs:142:14
  16:     0x55d2d92f5831 - <rustc_middle[7c51428e7903af24]::ty::Ty as rustdoc[863d67c12390cb82]::clean::Clean<rustdoc[863d67c12390cb82]::clean::types::Type>>::clean
  17:     0x55d2d94ed3b7 - <&mut rustdoc[863d67c12390cb82]::clean::utils::substs_to_args::{closure#0} as core[864d7ae58f623181]::ops::function::FnMut<(&rustc_middle[7c51428e7903af24]::ty::subst::GenericArg,)>>::call_mut
  18:     0x55d2d93910ee - <alloc[642b2cd0ec1e1a81]::vec::Vec<rustdoc[863d67c12390cb82]::clean::types::GenericArg> as alloc[642b2cd0ec1e1a81]::vec::spec_from_iter::SpecFromIter<rustdoc[863d67c12390cb82]::clean::types::GenericArg, core[864d7ae58f623181]::iter::adapters::filter_map::FilterMap<core[864d7ae58f623181]::slice::iter::Iter<rustc_middle[7c51428e7903af24]::ty::subst::GenericArg>, rustdoc[863d67c12390cb82]::clean::utils::substs_to_args::{closure#0}>>>::from_iter
  19:     0x55d2d94f09ec - rustdoc[863d67c12390cb82]::clean::utils::external_path
  20:     0x55d2d943884a - rustdoc[863d67c12390cb82]::clean::clean_trait_ref_with_bindings
  21:     0x55d2d9438fe7 - rustdoc[863d67c12390cb82]::clean::clean_poly_trait_ref_with_bindings
  22:     0x55d2d94319f5 - <&mut <rustc_middle[7c51428e7903af24]::ty::Ty as rustdoc[863d67c12390cb82]::clean::Clean<rustdoc[863d67c12390cb82]::clean::types::Type>>::clean::{closure#3} as core[864d7ae58f623181]::ops::function::FnMut<(&rustc_middle[7c51428e7903af24]::ty::Predicate,)>>::call_mut
  23:     0x55d2d9391bfe - <alloc[642b2cd0ec1e1a81]::vec::Vec<rustdoc[863d67c12390cb82]::clean::types::GenericBound> as alloc[642b2cd0ec1e1a81]::vec::spec_from_iter::SpecFromIter<rustdoc[863d67c12390cb82]::clean::types::GenericBound, core[864d7ae58f623181]::iter::adapters::filter_map::FilterMap<core[864d7ae58f623181]::slice::iter::Iter<rustc_middle[7c51428e7903af24]::ty::Predicate>, <rustc_middle[7c51428e7903af24]::ty::Ty as rustdoc[863d67c12390cb82]::clean::Clean<rustdoc[863d67c12390cb82]::clean::types::Type>>::clean::{closure#3}>>>::from_iter
  24:     0x55d2d92f496f - <rustc_middle[7c51428e7903af24]::ty::Ty as rustdoc[863d67c12390cb82]::clean::Clean<rustdoc[863d67c12390cb82]::clean::types::Type>>::clean
  25:     0x55d2d953dd6f - <rustdoc[863d67c12390cb82]::core::DocContext>::with_param_env::<alloc[642b2cd0ec1e1a81]::vec::Vec<rustdoc[863d67c12390cb82]::clean::types::Item>, rustdoc[863d67c12390cb82]::clean::clean_maybe_renamed_item::{closure#1}>
  26:     0x55d2d93885a5 - <alloc[642b2cd0ec1e1a81]::vec::Vec<rustdoc[863d67c12390cb82]::clean::types::Item> as alloc[642b2cd0ec1e1a81]::vec::spec_extend::SpecExtend<rustdoc[863d67c12390cb82]::clean::types::Item, core[864d7ae58f623181]::iter::adapters::flatten::FlatMap<core[864d7ae58f623181]::slice::iter::Iter<(&rustc_hir[aeeaacae47778f8d]::hir::Item, core[864d7ae58f623181]::option::Option<rustc_span[ff646084812721ee]::symbol::Symbol>)>, alloc[642b2cd0ec1e1a81]::vec::Vec<rustdoc[863d67c12390cb82]::clean::types::Item>, <rustdoc[863d67c12390cb82]::visit_ast::Module as rustdoc[863d67c12390cb82]::clean::Clean<rustdoc[863d67c12390cb82]::clean::types::Item>>::clean::{closure#2}>>>::spec_extend
  27:     0x55d2d9437fe6 - <rustdoc[863d67c12390cb82]::visit_ast::Module as rustdoc[863d67c12390cb82]::clean::Clean<rustdoc[863d67c12390cb82]::clean::types::Item>>::clean
  28:     0x55d2d94f0083 - rustdoc[863d67c12390cb82]::clean::utils::krate
  29:     0x55d2d955f808 - rustdoc[863d67c12390cb82]::main_options::{closure#0}::{closure#0}::{closure#1}
  30:     0x55d2d94ff889 - <rustc_interface[dacacaf3651fe347]::passes::QueryContext>::enter::<rustdoc[863d67c12390cb82]::main_options::{closure#0}::{closure#0}::{closure#1}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>
  31:     0x55d2d944de14 - <rustc_interface[dacacaf3651fe347]::interface::Compiler>::enter::<rustdoc[863d67c12390cb82]::main_options::{closure#0}::{closure#0}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>
  32:     0x55d2d92cb85a - rustc_span[ff646084812721ee]::with_source_map::<core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>, rustc_interface[dacacaf3651fe347]::interface::create_compiler_and_run<core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>, rustdoc[863d67c12390cb82]::main_options::{closure#0}>::{closure#1}>
  33:     0x55d2d9475c44 - rustc_interface[dacacaf3651fe347]::interface::create_compiler_and_run::<core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>, rustdoc[863d67c12390cb82]::main_options::{closure#0}>
  34:     0x55d2d92ce6d7 - <scoped_tls[d6203f3b3010b98e]::ScopedKey<rustc_span[ff646084812721ee]::SessionGlobals>>::set::<rustdoc[863d67c12390cb82]::main_args::{closure#0}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>
  35:     0x55d2d9478aff - std[fd2a1eaf3e081d4d]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[dacacaf3651fe347]::util::run_in_thread_pool_with_globals<rustdoc[863d67c12390cb82]::main_args::{closure#0}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>::{closure#0}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>
  36:     0x55d2d9521d89 - <<std[fd2a1eaf3e081d4d]::thread::Builder>::spawn_unchecked_<rustc_interface[dacacaf3651fe347]::util::run_in_thread_pool_with_globals<rustdoc[863d67c12390cb82]::main_args::{closure#0}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>::{closure#0}, core[864d7ae58f623181]::result::Result<(), rustc_errors[14f27e81a88a2c7]::ErrorGuaranteed>>::{closure#1} as core[864d7ae58f623181]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  37:     0x7f242c6aabf3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h211dae6d8dec3611
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/alloc/src/boxed.rs:1866:9
  38:     0x7f242c6aabf3 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb54334e8a7bb1f12
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/alloc/src/boxed.rs:1866:9
  39:     0x7f242c6aabf3 - std::sys::unix::thread::Thread::new::thread_start::hf8b57655f2f2e68e
                               at /rustc/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/std/src/sys/unix/thread.rs:108:17
  40:     0x7f242c39b5c2 - start_thread
  41:     0x7f242c420584 - __clone
  42:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.62.0-nightly (4ca19e09d 2022-04-19) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
error: aborting due to previous error
```

</p>
</details>

I think I have a smaller reproducer for the same bug, or at least one that leads to the same error. This one doesn't require any unstable feature and crashes both on 1.63 and on 1.66.0-nightly (432abd86f 2022-09-20) running on x86_64-unknown-linux-gnu.

Note that my reproducer is not expected to compile, it shouldn't type-check. But it should cause a panic either.

```rust
trait A<T> {}

struct B<C: ?Sized, D: A> {} 
```
# Meta

```
rustdoc 1.66.0-nightly (432abd86f 2022-09-20)
binary: rustdoc
commit-hash: 432abd86f231c908f6df3cdd779e83f35084be90
commit-date: 2022-09-20
host: x86_64-unknown-linux-gnu
release: 1.66.0-nightly
LLVM version: 15.0.0
```

```
error[E0107]: missing generics for trait `A`
 --> src/lib.rs:3:24
  |
3 | struct B<C: ?Sized, D: A> {}
  |                        ^ expected 1 generic argument
  |
note: trait defined here, with 1 generic parameter: `T`
 --> src/lib.rs:1:7
  |
1 | trait A<T> {}
  |       ^ -
help: add missing generic argument
  |
3 | struct B<C: ?Sized, D: A<T>> {}
  |                        ~~~~

thread 'rustc' panicked at 'Error', src/librustdoc/clean/mod.rs:1737:25
stack backtrace:
   0:     0x7ffa1bbead50 - std::backtrace_rs::backtrace::libunwind::trace::h22157d54c25d05ad
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   1:     0x7ffa1bbead50 - std::backtrace_rs::backtrace::trace_unsynchronized::hf9a98b3986ed5962
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7ffa1bbead50 - std::sys_common::backtrace::_print_fmt::h44f8a801ddb575b4
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x7ffa1bbead50 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h306abe8d5f5fb26d
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x7ffa1bc45aae - core::fmt::write::h5a3c14feb177cc6f
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/core/src/fmt/mod.rs:1202:17
   5:     0x7ffa1bbdb865 - std::io::Write::write_fmt::h2fac6c752c3fce7e
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/io/mod.rs:1679:15
   6:     0x7ffa1bbeda03 - std::sys_common::backtrace::_print::h17490259a7578aec
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x7ffa1bbeda03 - std::sys_common::backtrace::print::h6dba4ce5923cf00b
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x7ffa1bbeda03 - std::panicking::default_hook::{{closure}}::h25daedc6802fa9d1
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/panicking.rs:295:22
   9:     0x7ffa1bbed6ef - std::panicking::default_hook::hffec061dc0fa1a5a
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/panicking.rs:314:9
  10:     0x7ffa1e3fc951 - <rustc_driver[2817d6214121e090]::DEFAULT_HOOK::{closure#0}::{closure#0} as core[7a17016bf310e64d]::ops::function::FnOnce<(&core[7a17016bf310e64d]::panic::panic_info::PanicInfo,)>>::call_once::{shim:vtable#0}
  11:     0x7ffa1bbee23d - std::panicking::rust_panic_with_hook::h74e697c31d32505c
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/panicking.rs:702:17
  12:     0x7ffa1bbee051 - std::panicking::begin_panic_handler::{{closure}}::hd0b9cdbd524ed365
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/panicking.rs:586:13
  13:     0x7ffa1bbeb1fc - std::sys_common::backtrace::__rust_end_short_backtrace::h3feb3f4e0604d329
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/sys_common/backtrace.rs:138:18
  14:     0x7ffa1bbeddb2 - rust_begin_unwind
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/panicking.rs:584:5
  15:     0x7ffa1bc42683 - core::panicking::panic_fmt::h0f41ee9e7f6526dd
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/core/src/panicking.rs:142:14
  16:     0x55def3e243ac - rustdoc[a4cb4ff8fe5b9369]::clean::clean_middle_ty
  17:     0x55def3c0268d - <&mut rustdoc[a4cb4ff8fe5b9369]::clean::utils::substs_to_args::{closure#0} as core[7a17016bf310e64d]::ops::function::FnMut<(&rustc_middle[e9ef365c3785ee8]::ty::subst::GenericArg,)>>::call_mut
  18:     0x55def3c4665e - <alloc[e7766d434969df5]::vec::Vec<rustdoc[a4cb4ff8fe5b9369]::clean::types::GenericArg> as alloc[e7766d434969df5]::vec::spec_extend::SpecExtend<rustdoc[a4cb4ff8fe5b9369]::clean::types::GenericArg, core[7a17016bf310e64d]::iter::adapters::filter_map::FilterMap<core[7a17016bf310e64d]::slice::iter::Iter<rustc_middle[e9ef365c3785ee8]::ty::subst::GenericArg>, rustdoc[a4cb4ff8fe5b9369]::clean::utils::substs_to_args::{closure#0}>>>::spec_extend
  19:     0x55def3c0b4f8 - rustdoc[a4cb4ff8fe5b9369]::clean::utils::external_path
  20:     0x55def3e19c09 - rustdoc[a4cb4ff8fe5b9369]::clean::clean_trait_ref_with_bindings
  21:     0x55def3e1a0de - rustdoc[a4cb4ff8fe5b9369]::clean::clean_poly_trait_ref_with_bindings
  22:     0x55def3e1aa93 - rustdoc[a4cb4ff8fe5b9369]::clean::clean_predicate
  23:     0x55def3c53fcf - <alloc[e7766d434969df5]::vec::Vec<rustdoc[a4cb4ff8fe5b9369]::clean::types::WherePredicate> as alloc[e7766d434969df5]::vec::spec_from_iter::SpecFromIter<rustdoc[a4cb4ff8fe5b9369]::clean::types::WherePredicate, core[7a17016bf310e64d]::iter::adapters::flatten::FlatMap<alloc[e7766d434969df5]::vec::into_iter::IntoIter<&rustc_middle[e9ef365c3785ee8]::ty::Predicate>, core[7a17016bf310e64d]::option::Option<rustdoc[a4cb4ff8fe5b9369]::clean::types::WherePredicate>, rustdoc[a4cb4ff8fe5b9369]::clean::clean_ty_generics::{closure#3}>>>::from_iter
  24:     0x55def3e1d041 - rustdoc[a4cb4ff8fe5b9369]::clean::clean_ty_generics
  25:     0x55def3bd140e - <rustdoc[a4cb4ff8fe5b9369]::clean::auto_trait::AutoTraitFinder>::param_env_to_generics
  26:     0x55def3c80067 - <rustc_trait_selection[df340bb754643847]::traits::auto_trait::AutoTraitFinder>::find_auto_trait_generics::<rustdoc[a4cb4ff8fe5b9369]::clean::types::Generics, <rustdoc[a4cb4ff8fe5b9369]::clean::auto_trait::AutoTraitFinder>::generate_for_trait::{closure#0}>
  27:     0x55def3bce2a6 - <rustdoc[a4cb4ff8fe5b9369]::clean::auto_trait::AutoTraitFinder>::generate_for_trait
  28:     0x55def3c67b51 - <alloc[e7766d434969df5]::vec::Vec<rustdoc[a4cb4ff8fe5b9369]::clean::types::Item> as alloc[e7766d434969df5]::vec::spec_from_iter::SpecFromIter<rustdoc[a4cb4ff8fe5b9369]::clean::types::Item, core[7a17016bf310e64d]::iter::adapters::filter_map::FilterMap<alloc[e7766d434969df5]::vec::into_iter::IntoIter<rustc_span[8a07e301efc44469]::def_id::DefId>, <rustdoc[a4cb4ff8fe5b9369]::clean::auto_trait::AutoTraitFinder>::get_auto_trait_impls::{closure#0}>>>::from_iter
  29:     0x55def3bcf002 - <rustdoc[a4cb4ff8fe5b9369]::clean::auto_trait::AutoTraitFinder>::get_auto_trait_impls
  30:     0x55def3c0ddd1 - rustdoc[a4cb4ff8fe5b9369]::clean::utils::get_auto_trait_and_blanket_impls
  31:     0x55def3dc1774 - <rustdoc[a4cb4ff8fe5b9369]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[a4cb4ff8fe5b9369]::visit::DocVisitor>::visit_item
  32:     0x55def3dc18ba - <rustdoc[a4cb4ff8fe5b9369]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[a4cb4ff8fe5b9369]::visit::DocVisitor>::visit_item
  33:     0x55def3dc27ed - <rustdoc[a4cb4ff8fe5b9369]::passes::collect_trait_impls::SyntheticImplCollector as rustdoc[a4cb4ff8fe5b9369]::visit::DocVisitor>::visit_crate
  34:     0x55def3e3b6e2 - <rustc_session[920c8e110a6e000f]::session::Session>::time::<alloc[e7766d434969df5]::vec::Vec<rustdoc[a4cb4ff8fe5b9369]::clean::types::Item>, rustdoc[a4cb4ff8fe5b9369]::passes::collect_trait_impls::collect_trait_impls::{closure#0}>
  35:     0x55def3dbf602 - rustdoc[a4cb4ff8fe5b9369]::passes::collect_trait_impls::collect_trait_impls
  36:     0x55def3e3bcbd - <rustc_session[920c8e110a6e000f]::session::Session>::time::<rustdoc[a4cb4ff8fe5b9369]::clean::types::Crate, rustdoc[a4cb4ff8fe5b9369]::core::run_global_ctxt::{closure#7}>
  37:     0x55def3d144b8 - rustdoc[a4cb4ff8fe5b9369]::core::run_global_ctxt
  38:     0x55def3e3bf9f - <rustc_session[920c8e110a6e000f]::session::Session>::time::<(rustdoc[a4cb4ff8fe5b9369]::clean::types::Crate, rustdoc[a4cb4ff8fe5b9369]::config::RenderOptions, rustdoc[a4cb4ff8fe5b9369]::formats::cache::Cache), rustdoc[a4cb4ff8fe5b9369]::main_options::{closure#0}::{closure#0}::{closure#1}::{closure#0}>
  39:     0x55def3d86531 - <rustc_interface[2d255db050a36d31]::passes::QueryContext>::enter::<rustdoc[a4cb4ff8fe5b9369]::main_options::{closure#0}::{closure#0}::{closure#1}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>
  40:     0x55def3b6c8ea - <rustc_interface[2d255db050a36d31]::interface::Compiler>::enter::<rustdoc[a4cb4ff8fe5b9369]::main_options::{closure#0}::{closure#0}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>
  41:     0x55def3eaf450 - rustc_span[8a07e301efc44469]::with_source_map::<core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>, rustc_interface[2d255db050a36d31]::interface::create_compiler_and_run<core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>, rustdoc[a4cb4ff8fe5b9369]::main_options::{closure#0}>::{closure#1}>
  42:     0x55def3baf965 - rustdoc[a4cb4ff8fe5b9369]::main_options
  43:     0x55def3d842cb - <scoped_tls[f167104904068c68]::ScopedKey<rustc_span[8a07e301efc44469]::SessionGlobals>>::set::<rustdoc[a4cb4ff8fe5b9369]::main_args::{closure#0}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>
  44:     0x55def3e518c0 - std[7414053366649774]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[2d255db050a36d31]::util::run_in_thread_pool_with_globals<rustdoc[a4cb4ff8fe5b9369]::main_args::{closure#0}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>::{closure#0}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>
  45:     0x55def3d01b5c - <<std[7414053366649774]::thread::Builder>::spawn_unchecked_<rustc_interface[2d255db050a36d31]::util::run_in_thread_pool_with_globals<rustdoc[a4cb4ff8fe5b9369]::main_args::{closure#0}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>::{closure#0}, core[7a17016bf310e64d]::result::Result<(), rustc_errors[aa542ed4628157d3]::ErrorGuaranteed>>::{closure#1} as core[7a17016bf310e64d]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  46:     0x7ffa1bbf8013 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h59c19dbfda82d3be
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/alloc/src/boxed.rs:1940:9
  47:     0x7ffa1bbf8013 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h5cc1cabf17fe5922
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/alloc/src/boxed.rs:1940:9
  48:     0x7ffa1bbf8013 - std::sys::unix::thread::Thread::new::thread_start::h8f3eeb92c9304ef6
                               at /rustc/432abd86f231c908f6df3cdd779e83f35084be90/library/std/src/sys/unix/thread.rs:108:17
  49:     0x7ffa1b95974d - <unknown>
  50:     0x7ffa1b9db700 - <unknown>
  51:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.66.0-nightly (432abd86f 2022-09-20) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
error: aborting due to previous error

For more information about this error, try `rustc --explain E0107`.
```

--- a/src/librustdoc/clean/mod.rs
+++ b/src/librustdoc/clean/mod.rs
@@ -1853,7 +1853,7 @@ pub(crate) fn clean_middle_ty<'tcx>(
         ty::Placeholder(..) => panic!("Placeholder"),
         ty::GeneratorWitness(..) => panic!("GeneratorWitness"),
         ty::Infer(..) => panic!("Infer"),
-        ty::Error(_) => panic!("Error"),
+        ty::Error(_) => rustc_errors::FatalError.raise(),
     }
 }
 

--- a/src/test/rustdoc-ui/issue-105334.rs
+++ b/src/test/rustdoc-ui/issue-105334.rs
@@ -1,0 +1,2 @@
+impl Vec< br##"*.."## > {}
+//~^ ERROR

--- a/src/test/rustdoc-ui/issue-105334.stderr
+++ b/src/test/rustdoc-ui/issue-105334.stderr
@@ -1,0 +1,9 @@
+error[E0747]: constant provided when a type was expected
+  --> $DIR/issue-105334.rs:1:11
+   |
+LL | impl Vec< br##"*.."## > {}
+   |           ^^^^^^^^^^^
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0747`.

--- a/src/test/rustdoc-ui/issue-105737.rs
+++ b/src/test/rustdoc-ui/issue-105737.rs
@@ -1,0 +1,4 @@
+impl Vec<lol> {}
+//~^ ERROR
+
+pub fn lol() {}

--- a/src/test/rustdoc-ui/issue-105737.stderr
+++ b/src/test/rustdoc-ui/issue-105737.stderr
@@ -1,0 +1,12 @@
+error[E0747]: constant provided when a type was expected
+  --> $DIR/issue-105737.rs:1:10
+   |
+LL | impl Vec<lol> {}
+   |          ^^^
+   |
+   = help: `lol` is a function item, not a type
+   = help: function item types cannot be named directly
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0747`.

--- a/src/test/rustdoc-ui/issue-105742.rs
+++ b/src/test/rustdoc-ui/issue-105742.rs
@@ -1,0 +1,40 @@
+// compile-flags: -Znormalize-docs
+
+use std::ops::Index;
+
+pub fn next<'a, T>(s: &'a mut dyn SVec<Item = T, Output = T>) {
+//~^ ERROR
+//~^^ ERROR
+//~^^^ ERROR
+    let _ = s;
+}
+
+pub trait SVec: Index<
+    <Self as SVec>::Item,
+//~^ ERROR
+//~^^ ERROR
+//~^^^ ERROR
+//~^^^^ ERROR
+    Output = <Index<<Self as SVec>::Item,
+//~^ ERROR
+//~^^ ERROR
+//~^^^ ERROR
+//~^^^^ ERROR
+    Output = <Self as SVec>::Item> as SVec>::Item,
+//~^ ERROR
+//~^^ ERROR
+//~^^^ ERROR
+//~^^^^ ERROR
+//~^^^^^ ERROR
+//~^^^^^^ ERROR
+//~^^^^^^^ ERROR
+//~^^^^^^^^ ERROR
+> {
+    type Item<'a, T>;
+
+    fn len(&self) -> <Self as SVec>::Item;
+    //~^ ERROR
+    //~^^ ERROR
+    //~^^^ ERROR
+    //~^^^^ ERROR
+}

--- a/src/test/rustdoc-ui/issue-105742.stderr
+++ b/src/test/rustdoc-ui/issue-105742.stderr
@@ -1,0 +1,385 @@
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:13:21
+   |
+LL |     <Self as SVec>::Item,
+   |                     ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     <Self as SVec>::Item<'a>,
+   |                     ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:13:21
+   |
+LL |     <Self as SVec>::Item,
+   |                     ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     <Self as SVec>::Item<T>,
+   |                     ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:18:37
+   |
+LL |     Output = <Index<<Self as SVec>::Item,
+   |                                     ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     Output = <Index<<Self as SVec>::Item<'a>,
+   |                                     ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:18:37
+   |
+LL |     Output = <Index<<Self as SVec>::Item,
+   |                                     ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     Output = <Index<<Self as SVec>::Item<T>,
+   |                                     ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:30
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                              ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     Output = <Self as SVec>::Item<'a>> as SVec>::Item,
+   |                              ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:30
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                              ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     Output = <Self as SVec>::Item<T>> as SVec>::Item,
+   |                              ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:46
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                                              ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item<'a>,
+   |                                              ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:46
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                                              ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item<T>,
+   |                                              ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:5:40
+   |
+LL | pub fn next<'a, T>(s: &'a mut dyn SVec<Item = T, Output = T>) {
+   |                                        ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL | pub fn next<'a, T>(s: &'a mut dyn SVec<Item<'_> = T, Output = T>) {
+   |                                        ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:5:40
+   |
+LL | pub fn next<'a, T>(s: &'a mut dyn SVec<Item = T, Output = T>) {
+   |                                        ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL | pub fn next<'a, T>(s: &'a mut dyn SVec<Item<T> = T, Output = T>) {
+   |                                        ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:13:21
+   |
+LL |     <Self as SVec>::Item,
+   |                     ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     <Self as SVec>::Item<'a>,
+   |                     ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:13:21
+   |
+LL |     <Self as SVec>::Item,
+   |                     ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     <Self as SVec>::Item<T>,
+   |                     ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:18:37
+   |
+LL |     Output = <Index<<Self as SVec>::Item,
+   |                                     ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     Output = <Index<<Self as SVec>::Item<'a>,
+   |                                     ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:18:37
+   |
+LL |     Output = <Index<<Self as SVec>::Item,
+   |                                     ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     Output = <Index<<Self as SVec>::Item<T>,
+   |                                     ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:30
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                              ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     Output = <Self as SVec>::Item<'a>> as SVec>::Item,
+   |                              ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:30
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                              ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     Output = <Self as SVec>::Item<T>> as SVec>::Item,
+   |                              ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:46
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                                              ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item<'a>,
+   |                                              ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:23:46
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item,
+   |                                              ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     Output = <Self as SVec>::Item> as SVec>::Item<T>,
+   |                                              ~~~~~~~
+
+error[E0038]: the trait `SVec` cannot be made into an object
+  --> $DIR/issue-105742.rs:5:31
+   |
+LL | pub fn next<'a, T>(s: &'a mut dyn SVec<Item = T, Output = T>) {
+   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `SVec` cannot be made into an object
+   |
+note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
+  --> $DIR/issue-105742.rs:12:17
+   |
+LL |    pub trait SVec: Index<
+   |  ____________----__^
+   | |            |
+   | |            this trait cannot be made into an object...
+LL | |      <Self as SVec>::Item,
+LL | |
+LL | |
+...  |
+LL | |/     Output = <Index<<Self as SVec>::Item,
+LL | ||
+LL | ||
+LL | ||
+LL | ||
+LL | ||     Output = <Self as SVec>::Item> as SVec>::Item,
+   | ||_________________________________________________^ ...because it uses `Self` as a type parameter
+...  |
+LL | |
+LL | |  > {
+   | |__^ ...because it uses `Self` as a type parameter
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:35:38
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item;
+   |                                      ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item<'_>;
+   |                                      ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:35:38
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item;
+   |                                      ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item<T>;
+   |                                      ~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:35:38
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item;
+   |                                      ^^^^ expected 1 lifetime argument
+   |
+note: associated type defined here, with 1 lifetime parameter: `'a`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^ --
+help: add missing lifetime argument
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item<'_>;
+   |                                      ~~~~~~~~
+
+error[E0107]: missing generics for associated type `SVec::Item`
+  --> $DIR/issue-105742.rs:35:38
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item;
+   |                                      ^^^^ expected 1 generic argument
+   |
+note: associated type defined here, with 1 generic parameter: `T`
+  --> $DIR/issue-105742.rs:33:10
+   |
+LL |     type Item<'a, T>;
+   |          ^^^^     -
+help: add missing generic argument
+   |
+LL |     fn len(&self) -> <Self as SVec>::Item<T>;
+   |                                      ~~~~~~~
+
+error: aborting due to 23 previous errors
+
+Some errors have detailed explanations: E0038, E0107.
+For more information about an error, try `rustc --explain E0038`.

--- a/src/test/rustdoc-ui/issue-106226.rs
+++ b/src/test/rustdoc-ui/issue-106226.rs
@@ -1,0 +1,3 @@
+// This is a regression test for <https://github.com/rust-lang/rust/issues/106226>.
+type F = [_; ()];
+//~^ ERROR

--- a/src/test/rustdoc-ui/issue-106226.stderr
+++ b/src/test/rustdoc-ui/issue-106226.stderr
@@ -1,0 +1,9 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-106226.rs:2:14
+   |
+LL | type F = [_; ()];
+   |              ^^ expected `usize`, found `()`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0308`.

--- a/src/test/rustdoc-ui/issue-96287.rs
+++ b/src/test/rustdoc-ui/issue-96287.rs
@@ -1,0 +1,17 @@
+#![feature(type_alias_impl_trait)]
+
+pub trait TraitWithAssoc {
+    type Assoc;
+}
+
+pub type Foo<V> = impl Trait<V::Assoc>;
+//~^ ERROR
+//~^^ ERROR
+
+pub trait Trait<U> {}
+
+impl<W> Trait<W> for () {}
+
+pub fn foo_desugared<T: TraitWithAssoc>(_: T) -> Foo<T> {
+    ()
+}

--- a/src/test/rustdoc-ui/issue-96287.stderr
+++ b/src/test/rustdoc-ui/issue-96287.stderr
@@ -1,0 +1,15 @@
+error[E0220]: associated type `Assoc` not found for `V`
+  --> $DIR/issue-96287.rs:7:33
+   |
+LL | pub type Foo<V> = impl Trait<V::Assoc>;
+   |                                 ^^^^^ there is a similarly named associated type `Assoc` in the trait `TraitWithAssoc`
+
+error[E0220]: associated type `Assoc` not found for `V`
+  --> $DIR/issue-96287.rs:7:33
+   |
+LL | pub type Foo<V> = impl Trait<V::Assoc>;
+   |                                 ^^^^^ there is a similarly named associated type `Assoc` in the trait `TraitWithAssoc`
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0220`.

